### DISCUSSION TITLE
Default mod combination

### DISCUSSION THREAD

**Discussion by developomp** [ppy/osu#13198] [Ideas]
**Describe the new feature:** 
As of the moment, pressing the "Deselect All" button in the mod menu removes all applied mod and "resets" any modification made to the difficulty of the beatmap. Though this is more than enough for some, many people have their own preferred mod combinations. Some people even play with specific mod combinations exclusively. It's a pain having to enable these mods every session. That's why I suggest adding a "default mod" settings in osu!lazer.

Here's my proposal:

- Right-clicking a mod after setting it to a certain value makes it the default mod
- Right-clicking it again resets the default mod except when another mod variant is selected (read for more info)
- Now pressing the "Deselect All" button will change the value to whatever is set as default
- When a default mod is set, there will be a green dot in the upper right corner of that mod. This will turn dark if another mod variant is set. (e.g. when DT is the default mod but NC is selected)

I intentionally designed this feature to be as easy to implement as possible so if you (other people) have any suggestions to make it more intuitive, please do post them in this issue thread.

**Proposal designs of the feature:** 
Normal:
![1](https://user-images.githubusercontent.com/40858122/117974314-648f0f80-b368-11eb-9873-7be00a6a472a.png)

After selecting a mod as default mod:
![2](https://user-images.githubusercontent.com/40858122/117974317-65c03c80-b368-11eb-9f0b-4ebdb4f23b89.png)

When a default mod is set in another mod variant:
![3](https://user-images.githubusercontent.com/40858122/117974921-175f6d80-b369-11eb-87db-de23a4fde9e1.png)

**Comment by sr229** [comment]
if I remember there were mod presets coming (based on @arflyte's streams) which makes this proposal obsolete in a way.
